profiling/src/profiling/mod.rs

```diff
@@ -572,6 +572,12 @@ impl Profiler {
     }
 
     pub fn send_sample(&self, message: SampleMessage) -> Result<(), TrySendError<ProfilerMessage>> {
+        if message.key.sample_types.len() == 0 {
+            // profiling disabled, this should not happen!
+            warn!("You spot a bug in the profiler, please be so kind and report this do Datadog.");
+            // this return is technically not correct :-(
+            return Err(TrySendError::Disconnected(ProfilerMessage::Sample(message)));
+        }
```

Comment:
@morrisonlevi I am not really happy with having this check for every sample that we send (also the return value is incorrect). OTOH this is the perfect spot, as all samples are going through this. If you have some time, could we put some thought into this? One idea is to check in the other thread, so we move this out of the hot-path

Reply:
I moved the logic to the other thread